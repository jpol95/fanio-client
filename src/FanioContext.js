import React from 'react'

const FanioContext = React.createContext({
    fandomList: [],
    reviewList: [],
    sectionList: [], 
    subList: [],
    tagList: [],
    reviewTagList: [],
    installmentList: [],
    handleSubmitReview: () => {}, 
    handleAddFandom: () => {}, 
    handleSubmitInstallments: () => {}, 
    handleGetReview: () => {}
})

export default FanioContext