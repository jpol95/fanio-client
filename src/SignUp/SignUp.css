.basic{
    background-color: lightgray;
}

.personal, .fandom-view{
    margin-top: 20px;
    margin-bottom: 20px;
}
.personal p{
    margin-top: 0px;
}
.personal, .fandom, .fandom-view{
    background-color: lightgray;
    padding: 20px;
}

.personal{
   padding-bottom: 300px 
}

.personal .profile-picture{
    background-color: grey;
    display: flex;
    height:200px;
}

.personal .profile-picture p{
    margin: auto;
}