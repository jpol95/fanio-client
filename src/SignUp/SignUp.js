import React from "react";
import "./SignUp.css";

export default class Signup extends React.Component {
  render() {
    return (
      <form class="new-user-form">
        <div className="basic">
          <p> Basic account information </p>
          <label htmlFor="username">Username</label>
          <input id="username" type="text" />
          <label htmlFor="pw">Password</label>
          <input id="pw" type="text" />
          <label htmlFor="pw-confirm">Confirm Password</label>
          <input id="pw-confirm" type="text" />
        </div>

        <div className="personal">
          <p>Tell us about yourself</p>
          <label htmlFor="username">Name</label>
          <input id="username" type="text" />
          <label htmlFor="fandoms">Fandoms</label>
          <input id="fandoms" type="text" />
          <label htmlFor="hobbies">Hobbies</label>
          <input id="hobbies" type="text" />
          <label htmlFor="interests">Interests</label>
          <input id="interests" type="text" />
          <label htmlFor="city">City</label>
          <input id="city" type="text" />
          <label htmlFor="education">Education</label>
          <input id="education" type="text" />
        </div>
        <button type="submit">Create Profile</button>
      </form>
    );
  }
}
