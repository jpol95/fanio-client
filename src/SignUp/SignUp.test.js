import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SignUp from "./SignUp";
import FanioContext from "../FanioContext";
import dummyStore from "../dummy-store";
import { BrowserRouter } from "react-router-dom";

describe("SignUp component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <FanioContext.Provider
          value={{
            reviewList: dummyStore.reviewList,
            fandomList: dummyStore.fandomList,
            installmentList: dummyStore.installmentList,
            loggedInUser: 1,
            sectionList: dummyStore.sectionList,
          }}
        >
          {" "}
          <SignUp />{" "}
        </FanioContext.Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <FanioContext.Provider
            value={{
              reviewList: dummyStore.reviewList,
              fandomList: dummyStore.fandomList,
              installmentList: dummyStore.installmentList,
              loggedInUser: 1,
              sectionList: dummyStore.sectionList,
            }}
          >
            {" "}
            <SignUp />{" "}
          </FanioContext.Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
