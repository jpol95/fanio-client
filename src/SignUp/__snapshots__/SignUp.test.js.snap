// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders the UI as expected 1`] = `
Array [
  " ",
  <form
    className="new-user-form"
    onSubmit={[Function]}
  >
    <div
      className="basic"
    >
      <p>
         Basic account information 
      </p>
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        id="username"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="pw"
      >
        Password
      </label>
      <input
        id="pw"
        onChange={[Function]}
        type="password"
      />
      <label
        htmlFor="pw-confirm"
      >
        Confirm Password
      </label>
      <input
        id="pw-confirm"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="personal"
    >
      <p>
        Tell us about yourself
      </p>
      <label
        htmlFor="fullname"
      >
        Full Name
      </label>
      <input
        id="fullname"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="interests"
      >
        Interests(comma separated list)
      </label>
      <input
        defaultValue=""
        id="interests"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="city"
      >
        City
      </label>
      <input
        defaultValue=""
        id="city"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="education"
      >
        Education
      </label>
      <input
        defaultValue=""
        id="education"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="submit-account-button"
      disabled={
        <div
          className="error"
        >
          Password must be longer than 8 characters 
        </div>
      }
      type="submit"
    >
      Submit
    </button>
  </form>,
  " ",
]
`;
